docs/current_docs/cookbook/snippets/clone-git-repository/java/MyModule.java

```diff
@@ -0,0 +1,26 @@
+package io.dagger.modules.mymodule;
+
+import static io.dagger.client.Dagger.dag;
+
+import io.dagger.client.Container;
+import io.dagger.client.Directory;
+import io.dagger.client.GitRepository;
+import io.dagger.module.annotation.Function;
+import io.dagger.module.annotation.Object;
+
+@Object
+public class MyModule {
+  @Function
+  public Container clone(String repository, String locator, String ref) {
```

Comment:
This recipe has changed and there is a new version of this for SSH cloning of private repositories: https://github.com/dagger/dagger/pull/10092 and https://github.com/dagger/dagger/pull/9770